src/daft-connect/src/lib.rs

```diff
@@ -0,0 +1,463 @@
+        Err(unimplemented!(
+            "add_artifacts operation is not yet implemented"
+        ))
+    }
+
+    #[tracing::instrument(skip_all)]
+    async fn analyze_plan(
+        &self,
+        request: Request<AnalyzePlanRequest>,
+    ) -> Result<Response<AnalyzePlanResponse>, Status> {
+        use spark_connect::analyze_plan_request::*;
+        let request = request.into_inner();
+
+        let plan_readable = ron::ser::to_string_pretty(&request, pretty_config()).unwrap();
+        info!("got analyze plan\n{plan_readable}");
+
+        let AnalyzePlanRequest {
+            session_id,
+            analyze,
+            ..
+        } = request;
+
+        let Some(analyze) = analyze else {
+            return Err(Status::invalid_argument("analyze is required"));
+        };
+
+        match analyze {
+            Analyze::Schema(Schema { plan }) => {
+                let Some(Plan { op_type }) = plan else {
+                    return Err(Status::invalid_argument("plan is required"));
+                };
+
+                let Some(OpType::Root(relation)) = op_type else {
+                    return Err(Status::invalid_argument("op_type is required to be root"));
+                };
+
+                let result = convert::connect_schema(relation)?;
+
+                let schema = analyze_plan_response::DdlParse {
+                    parsed: Some(result),
+                };
+
+                let response = AnalyzePlanResponse {
+                    session_id,
+                    server_side_session_id: String::new(),
+                    result: Some(analyze_plan_response::Result::DdlParse(schema)),
+                };
+
+                println!("response: {response:#?}");
+
+                Ok(Response::new(response))
+            }
+            Analyze::TreeString(tree_string) => {
+                if let Some(level) = tree_string.level {
+                    warn!("Ignoring level {level} in TreeString");
+                }
+
+                let Some(plan) = tree_string.plan else {
+                    return Err(invalid_argument!("TreeString must have a plan"));
+                };
+
+                let Some(op_type) = plan.op_type else {
+                    return Err(invalid_argument!("plan must have an op_type"));
+                };
+
+                println!("op_type: {op_type:?}");
+
+                let OpType::Root(plan) = op_type else {
+                    return Err(invalid_argument!("Only op_type Root is supported"));
+                };
+
+                let logical_plan = match convert::to_logical_plan(plan) {
+                    Ok(lp) => lp,
+                    Err(e) => {
+                        return Err(invalid_argument!(
+                            "Failed to convert to logical plan: {e:?}"
+                        ));
+                    }
+                };
+
+                let logical_plan = logical_plan.build();
+
+                let res = std::thread::spawn(move || {
```

Comment:
why not `tokio::spawn` instead?

Reply:
it's an issue of the code creating its own runtime inside of a runtime